service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,217 @@
+      PaginateValidator paginateValidator,
+      TagDtoValidator tagDtoValidator) {
+    this.tagDao = tagDao;
+    this.mapper = mapper;
+    this.idValidator = idValidator;
+    this.paginateValidator = paginateValidator;
+    this.tagDtoValidator = tagDtoValidator;
+  }
+
+  @Override
+  @Transactional
+  public int findPaginated(int size) {
+    List<ErrorMessageDto> errors = paginateValidator.sizeValidate(size);
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors);
+    }
+    int countOfRecords = tagDao.findCountOfRecords();
+    int countPages = countOfRecords % size == 0 ? countOfRecords / size : countOfRecords / size + 1;
+    return countPages == 0 ? 1 : countPages;
+  }
+
+  @Override
+  @Transactional
+  public int findTagsOfCertificatePaginated(int size) {
+    List<ErrorMessageDto> errors = paginateValidator.sizeValidate(size);
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors);
+    }
+    int countOfRecords = tagDao.findCountOfRecords();
+    int countPages = countOfRecords % size == 0 ? countOfRecords / size : countOfRecords / size + 1;
+    return countPages == 0 ? 1 : countPages;
+  }
+
+  @Override
+  public List<TagDto> findAll(int page, int size, List<String> sortParams) {
+    List<ErrorMessageDto> errors = paginateValidator.sizeValidate(size);
+    errors.addAll(paginateValidator.pageValidate(page));
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors);
+    }
+    return tagDao.findAll(page, size, sortParams).stream()
+        .map(tag -> mapper.mapToDto(tag, false))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public List<TagDto> findByCertificateId(
+      Long certificateId, int size, int page, List<String> sortParams) {
+    List<ErrorMessageDto> errors = paginateValidator.sizeValidate(size);
+    errors.addAll(paginateValidator.pageValidate(page));
+    errors.addAll(idValidator.idValidate(certificateId));
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors);
+    }
+    return tagDao.findByCertificateId(certificateId, size, page, sortParams).stream()
+        .map(tag -> mapper.mapToDto(tag, false))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  @Transactional
+  public TagDto findById(Long id) {
+    List<ErrorMessageDto> errors = idValidator.idValidate(id);
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors);
+    }
+    Tag tag = tagDao.findById(id);
+    if (tag == null) {
+      throw new NotFoundException(
+          new ErrorMessageDto(
+              CustomHttpStatus.NotFound.TAG.getCode(),
+              LanguageManager.getMessage(LocaleMessages.TAG_NOT_FOUND).formatted(id)));
+    }
+    return mapper.mapToDto(tag, true);
+  }
+
+  @Override
+  @Transactional
+  public TagDto create(TagDto tagDto) {
+    List<ErrorMessageDto> errors = tagDtoValidator.tagDtoValidate(tagDto);
+    if (!errors.isEmpty()) {
+      throw new ValidationException(errors);
```

Comment:
@Exqzore Why Don't you throw Exception from validator? It will make your code cleaner.

Reply:
To get errors of all validators and display them to the client